### PR TITLE
where raises ArgumentError on unsupported argument types

### PR DESCRIPTION
**Pull request by jwworth**
This addresses the following issue:

'where doesn't warn for wrong values'
https://github.com/rails/rails/issues/20473

### CODE REVIEW

File: activerecord/lib/active_record/relation/query_methods.rb [hunk 1]
```diff
@@ -551,6 +551,8 @@ def where(opts = :chain, *rest)
         WhereChain.new(spawn)
       elsif opts.blank?
         self
+      elsif opts.is_a?(Numeric)
+        raise ArgumentError, 'wrong argument type (Numeric)'
```
Comment (senny) [inline activerecord/lib/active_record/relation/query_methods.rb]: would be nice to also include the number in the error message.

Comment (jwworth) [inline activerecord/lib/active_record/relation/query_methods.rb]: @senny makes sense; adding that change.

**Comment by chancancode** [comment]
?r @rafaelfranca

**Comment by sgrif** [comment]
I don't think that we should special case `Numeric`, we should raise on anything that we don't explicitly support.

r? me

**Comment by jwworth** [comment]
@sgrif I think the things we support here are Strings, Hashes, Arrays, and objects that respond to `to_h`, per:

https://github.com/rails/rails/blob/master/activemodel/lib/active_model/forbidden_attributes_protection.rb#L22

Does that make sense?

**Comment by sgrif** [comment]
sounds right

**Comment by jwworth** [comment]
@sgrif that makes sense; updated PR to raise unless argument is a supported type

**Comment by sgrif** [comment]
Cool. Can you squash?

**Comment by sgrif** [comment]
And also add a CHANGELOG entry

**Comment by jwworth** [comment]
@sgrif squashed with a CHANGELOG entry.

**Comment by matthewd** [comment]
It feels like this should be the job of the where clause factory to me ¯\\\_(ツ)_/¯

**Comment by sgrif** [comment]
Good point. I'll move it in a few

**Comment by jwworth** [comment]
@sgrif @matthewd Agreed. I can handle this today with another pull request, if you'd like

**Comment by sgrif** [comment]
All good, thanks. :)

**Pull request merged**